### PR TITLE
#05 AS: dockerized the app, written seeding scripts for db, updated CI to check docker compose up

### PR DESCRIPTION
**Pull request by AdoshSingh**
Dockerized the whole app.
Written unit tests for backend.
Written scripts for seeding database.
Updated CI, adding a check for docker compose up command.
Moved the markdown files from content to public directory since react uses that directory to serve static files in build version.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      cockroachdb:
+        condition: service_healthy
+
+  front_end:
+    build: ./frontend
+    ports:
+      - 4173:4173
+    depends_on:
+      api:
+        condition: service_healthy
```
Comment (dray92) [inline docker-compose.yml:43]:
```suggestion
  front_end:
    build: ./frontend
    ports:
      - 4173:4173
    depends_on:
      api:
        condition: service_healthy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:4173/"]
      interval: 10s
      timeout: 5s
      retries: 5
```

File: docker-compose-up.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+docker-compose up -d
```
Comment (dray92) [inline docker-compose-up.sh:3]: check my suggestion in the next section -- would that remove the need for this script?

File: backend/package.json [hunk 3]
```diff
@@ -2,13 +2,22 @@
   "name": "change-log-backend",
   "module": "index.ts",
   "type": "module",
+  "scripts": {
+    "start": "bun run scripts/db-init.ts && bun run scripts/db-seed.ts && bun run index.ts",
```
Comment (dray92) [inline backend/package.json:6]: wouldn't you want to do this as part of docker db init instead of everytime the app starts up?

Comment (dray92) [reply backend/package.json:6]: maybe even a separate script for `bootstrap` would do the trick?

File: docker-compose-up.sh (hunk 2 above)
Comment (AdoshSingh) [reply docker-compose-up.sh]: Added the suggested change in the frontend docker-compose.yml.
Removed the ```docker-compose-up.sh``` script, instead just added a script in CI pipeline to check the health of front_end container (its already checking health of backend and database, so no need to check them in CI).

File: backend/package.json (hunk 3 above)
Comment (AdoshSingh) [reply backend/package.json]: I've removed the db-init.ts and db-seed.ts files and am now using an init-db.sql file, which I provide directly to the database container. It executes the init-db.sql file just once upon initialization.

**Review by dray92 — APPROVED** [review]

**Pull request merged**